### PR TITLE
Audio stuff

### PR DESCRIPTION
**Pull request by nanaian**

### CODE REVIEW

File: src/code_e0b30_len_b80.c [hunk 1]
```diff
@@ -1,38 +1,217 @@
...
+/// @returns -1: no override; 0: override to variation 0; 1 override to variation 1
+s32 get_song_variation_override_for_cur_map(SongID songID) {
+    u32 i = 0;
+    Area* areas = gAreas;
+    SongID* allowed = gSongsUsingVariationFlag;
+    GameStatus** gameStatusPtr = gGameStatusPtr;
+
+    for (i = 0; i < ARRAY_COUNT(gSongsUsingVariationFlag); i++) {
+        if (allowed[i] == songID) {
+            GameStatus* gameStatus = *gameStatusPtr;
```
Comment (ethteck) [inline src/code_e0b30_len_b80.c:19]: You can't do `= GAME_STATUS` here?

Comment (ethteck) [reply src/code_e0b30_len_b80.c:19]: or above and reference it here? or try `GAME_STATUS->`

**Review by ethteck — DISMISSED** [review]

File: src/code_28910_len_5090.c [hunk 2]
```diff
@@ -1,16 +1,46 @@
...
+#include "code_28910_len_5090.h"
 
 INCLUDE_ASM(s32, "code_28910_len_5090", func_8004D510);
 
 INCLUDE_ASM(s32, "code_28910_len_5090", func_8004D794);
 
 INCLUDE_ASM(s32, "code_28910_len_5090", func_8004D7E0);
 
-INCLUDE_ASM(s32, "code_28910_len_5090", func_8004DA0C);
+s32 func_8004DA0C(UNK_TYPE arg0) {
```
Comment (pixel-stuck) [inline src/code_28910_len_5090.c:9]: this function may be able to be rewritten (potentially early return on the initial check of arg0)

File: src/code_e0b30_len_b80.c [hunk 3]
```diff
@@ -1,38 +1,215 @@
...
 
-INCLUDE_ASM(s32, "code_e0b30_len_b80", _set_music_track);
+s32 _set_music_track(s32 playerIndex, SongID songID, s32 variation, s32 fadeOutTime, s16 volume) {
+    GameStatus* gameStatus = GAME_STATUS;
+
+    if (gameStatus->demoState != 0) {
+        return 1;
+    } else {
+        // TODO: these two lines likely a MUSIC_PLAYER(idx) macro
+        MusicPlayer* musicPlayers = gMusicPlayers;
```
Comment (pixel-stuck) [inline src/code_e0b30_len_b80.c:56]: my guess is that this pattern might produce the correct codegen but isn't quite correct. Probably they have two instances of this particular line, one in each scope or so, then just always use musicPlayers[idx] or so. Maybe I'm wrong, though

File: src/code_28910_len_5090.c (hunk 2 above)
Comment (nanaian) [reply src/code_28910_len_5090.c:9]: I wasn't able to get this to match without a `ret` var- the asm stores in s0 then jumps to `move v0, s0; jr ra` which is weird

**Review by pixel-stuck — APPROVED** [review]

**Pull request merged**